### DISCUSSION TITLE
does RSC matters SEO for client component?

### DISCUSSION THREAD

**Discussion by dante01yoon** [vercel/next.js#72268] [App Router]
Greetings,

If we use ISR + client component, each revalidate time met, page would be rendered on request, client component will be also called on server, but there will be slot remains for client component in initial html document until react does hydration and fill the slot gap with RSC payload in script tag.

compare to before x-component content-type, Is this bad for SEO? Should search engine supports read x-component to give full optimization?

**Comment by icyJoseph** [comment]
The general idea is that, for bots, and other clients that don't support streaming, the entire page is served.

- https://github.com/vercel/next.js/issues/42063#issuecomment-1299022554

Though it depends, https://github.com/vercel/next.js/discussions/50829#discussioncomment-8364251 as pointed out here... what do we really mean with SEO... 

- https://github.com/vercel/next.js/discussions/43505


Also I should add that, without streaming involved, client components are just fine. As you mention, they are indeed SSR'd.

**Reply by dante01yoon** [reply]
Interesting. So, 
Either
 1.  client do not support http2 streaming, can' be indexed each page correctly,
 2.  it depends search engine allow script execution, which it definitely needs to be hydrated to get full render output.

**Reply by icyJoseph** [reply]
> client do not support http2 streaming, can' be indexed each page correctly,

Then HTTP 1.1 regular serving is used, AFAIK, so it will be fine. You can try this yourself with `curl` or whatever from your terminal

> it depends search engine allow script execution, which it definitely needs to be hydrated to get full render output.

This is where it can get a bit tricky... and where it can be hand wavy, because if the tooling is actually spinning a headless browser, to run the site, to crawl it and extract content... then yeah, but then streaming is one concern, among a whole bunch of concerns... it gets trickier.

**Reply by icyJoseph** [reply]
The tooling referenced here, https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming#seo should be helpful
